samples/net/http_server/README.rst

```diff
@@ -0,0 +1,342 @@
+   This configuration option sets the server port for the HTTP server sample when using IPv6.
+   The nRF91 modem does not support the SO_REUSEPORT socket option; therefore, separate ports
+   for IPv6 are specified to avoid errors when configuring the sockets.
+
+.. _CONFIG_HTTP_SERVER_SAMPLE_CLIENTS_MAX:
+
+CONFIG_HTTP_SERVER_SAMPLE_CLIENTS_MAX
+   This configuration option sets the maximum number of concurrent clients for the HTTP server sample.
+   Increasing this option impacts the server's performance and increases the
+   total memory footprint, as each client instance has its own thread with a dedicated stack.
+   The option sets the number of maximum concurrent clients per IP family, allowing a total of
+   4 clients to be connected simultaneously if both IPv4 and IPv6 are enabled and the value is set to 2.
+
+.. _CONFIG_HTTP_SERVER_SAMPLE_SERVER_CERTIFICATE_SEC_TAG:
+
+CONFIG_HTTP_SERVER_SAMPLE_SERVER_CERTIFICATE_SEC_TAG
+   This configuration option sets the security tag for the server certificate used in TLS.
+
+.. _CONFIG_HTTP_SERVER_SAMPLE_STACK_SIZE:
+
+CONFIG_HTTP_SERVER_SAMPLE_STACK_SIZE
+   This configuration option sets the stack size for the threads used in the HTTP server sample.
+
+.. _CONFIG_HTTP_SERVER_SAMPLE_RECEIVE_BUFFER_SIZE:
+
+CONFIG_HTTP_SERVER_SAMPLE_RECEIVE_BUFFER_SIZE
+   This configuration option sets the receive buffer size for the sockets used in the HTTP server sample.
+
+.. include:: /includes/wifi_credentials_shell.txt
+.. include:: /includes/wifi_credentials_static.txt
+
+Configuration files
+===================
+
+The sample includes pre-configured configuration files for the development kits that are supported:
+
+* :file:`prj.conf` - General configuration file for all devices.
+* :file:`boards/nrf7002dk_nrf5340_cpuapp_ns.conf` - Configuration file for the nRF7002 DK.
+* :file:`boards/nrf9161dk_nrf9161_ns.conf` - Configuration file for the nRF9161 DK.
+* :file:`boards/nrf9160dk_nrf9160_ns.conf` - Configuration file for the nRF9160 DK.
+* :file:`boards/thingy91_nrf9160_ns.conf` - Configuration file for the Thingy:91.
+
+Files that are located under the :file:`/boards` folder are automatically merged with the :file:`prj.conf` file when you build for the corresponding target.
+
+Building and running
+********************
+
+.. |sample path| replace:: :file:`samples/net/http_server`
+
+.. include:: /includes/build_and_run_ns.txt
+
+Testing
+=======
+
+|test_sample|
+
+1. |connect_kit|
+#. |connect_terminal|
+#. Reset your board.
+#. Observe that the board connects to the network and is waiting for incoming HTTP connections.
+#. Set/get the value of the two supported LEDs by performing REST calls to either of the two resource URLs. In the following example we are using `HTTPie`_:
+
+.. tabs::
+
+   .. tab:: Wi-Fi
+
+         Non TLS (unsecure):
+
+         .. code-block:: console
+
+            http PUT http://httpserver.local:80/led/1 --raw="1"
+            http GET http://httpserver.local:80/led/1
```

Comment:
Is there a way to enforce IPv6 with this command? If so, maybe it'd make sense to add a sample command for that case as well, given the server uses a different port for IPv6. Or at least add a comment about different port being used for IPv6 in the testing section, I think this info could be overlooked given it's only provided in the Kconfig seciton.

Reply:
Update the server to use the same port for IPv6, using the SO_REUSEADDR socketopt. Previously I got -112 without setting this option. EADDRINUSE

@jukkar Interesting, how would I do that?. I'm not sure if httpie supports forcing IPv6 other than explicitly using the IPv6 address directly in the request. Ill have to check that out.

Reply:
> Interesting, how would I do that?. I'm not sure if httpie supports forcing IPv6 other than explicitly using the IPv6 address directly in the request. Ill have to check that out.

See for example the `setup_tcp_socket()` in echo-service sample in upstream
https://github.com/zephyrproject-rtos/zephyr/blob/68bea801119fe9a479f4a53185dfa0c95d26ce95/samples/net/sockets/echo_service/src/main.c#L117
how it is done.
So `CONFIG_NET_IPV4_MAPPING_TO_IPV6=y` needs to be set and `IPV6_V6ONLY` socket option needs to be turned off. The system creates IPv6 socket only, but it allows IPv4 to connect to it.

Reply:
I don't believe our modem supports this feature I'm afraid which will cause some displacement between wifi and lte builds. Alternative is to offload tcp/ip to native but I think we should use the modems tcp/ip stack.

Reply:
Indeed, for offloaded case we cannot use this atm.

Reply:
@rlubos Even though I use the same ports now I still have issues using the hostname directly in the request for IPv6. Ill add some docs on this shortcoming and add some cmnds on how to use IPv6 address direclty insead, until its resolved. It might be that this is only on my network setup.